src/service/extensions-impl.js

```diff
@@ -578,16 +580,22 @@ export class Extensions {
   /**
    * Determine the need to add amp extension script to document.
    * @param {string} extensionId
+   * @param {string} version
    * @param {!ExtensionHolderDef} holder
    * @return {boolean}
    * @private
    */
-  isExtensionScriptRequired_(extensionId, holder) {
+  isExtensionScriptRequired_(extensionId, version, holder) {
     if (holder.loaded || holder.error) {
       return false;
     }
     if (holder.scriptPresent === undefined) {
-      const scriptsInHead = getExtensionScripts(this.win, extensionId);
+      const scriptsInHead = getExtensionScripts(
+        this.win,
+        extensionId,
+        version,
+        /* latest */ false
```

Comment:
Shouldn't we be requiring `latest` to be passed in?

Reply:
The problem is that `latest` is not known until the extension is actually loaded. There are only two APIs here that are called at that point: `registerExtension` and `reloadExtension`. All other APIs, including `preloadExtension`, `installExtensionForDoc`, etc - a caller doesn't actually know whether a particular version is latest or not.